### DISCUSSION TITLE
Micronaut client sends plain string instead of JSON when Content-Type: application/json request

### DISCUSSION THREAD

**Discussion by matadini** [micronaut-projects/micronaut-core#11401] [Q&A]
Hi everyone.

My problem/question/concern is: Let say I have a defined API (OpenAPI) like this:
```yml
  /api/v1/colors:
    get:
      tags: [ my-custom ]
      operationId: "getSelectedColor"
      responses:
        "200":
          description: "OK"
          content:
            application/json:
              schema:
                type: string
    post:
      tags: [ my-custom ]
      operationId: "selectColor"
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: string
      responses:
        "200":
          description: "OK"
```
As a result of generation I have interface:
```java
@Controller
public interface MyCustomApi {

    @Get("/api/v1/colors")
    String getSelectedColor();

    @Post("/api/v1/colors")
    void selectColor(
        @Body @NotNull String body
    );
}
```
Base on it, I implemented a controller. But, in tests where I am using a declarative client. Calling endpoint looks like this:
```txt
HTTP Client Response Received (200) for Request: GET http://localhost.localdomain:33895/api/v1/colors
date: Tue, 3 Dec 2024 17:20:19 GMT
content-type: application/json
content-length: 4
Response Body
----
BLUE
----

Sending HTTP POST to http://localhost.localdomain:33895/api/v1/colors
Content-Type: application/json
host: localhost.localdomain:33895
Request Body
----
BLUE
----
```

Base on `application/json` content type, I expected that request body and request result will be something like this `"BLUE"`.
I found a discussion with similar question:
https://github.com/micronaut-projects/micronaut-core/discussions/11118
And my question is: is it applicable for micronaut-client? I mean, in case of string, micronaut-client expects raw/encoded JSON?
Why IMHO it is a bug, because when I use other client (lets say POSTMAN/CURL/IntelliJ HTTP) and send something like this:
```
POST /api/v1/colors HTTP/1.1
Content-Type: application/json
Host: localhost:8080
Content-Length: 6

"BLUE"
```
and try do something like this:
```
    @Override
    public void selectColor(String body) {
        System.out.println(Color.valueOf(body));
        selected = body;
    }
```
I will receive:
```json
    "errors": [
      {
        "message": "Internal Server Error: No enum constant com.example.model.Color.\"BLUE\""
      }
```
In my situation, I cannot use `enum` as `setColor` argument because of:
https://github.com/micronaut-projects/micronaut-openapi/issues/1896

Thank you in advance for your response, explanation.

**Comment by yawkat** [comment]
@andriy-dmytruk maybe we need a change in the generator here. returning String directly will lead to raw body writing

**Reply by andriy-dmytruk** [reply]
To be honest, this behavior is confusing, as string is a json type, too.

In any case, what can I change the return type to in order to fix this?

**Reply by yawkat** [reply]
I think the only way is a type like `record StringWrapper(@JsonValue String s)`
